app/js/enhancedDate.js

```diff
@@ -1,5 +1,159 @@
 /* 
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+ * Date object: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date
+ *
+ *
+ * setDate: Stores the passed-in date. 
+ * Takes milliseconds after epoch or a date object. 
+ * Do some type-checking to make sure you have a valid type. 
+ * If no value is passed, default to now
+ *
+ * getDate: Returns either milliseconds after epoch by default 
+ * or a date object if true is passed as an argument.
+ *
+ * getDayName: Returns the full day name as a string, e.g. "Monday", "Tuesday", etc.
+ *
+ * getMonthName: Return the full month name as a string, e.g. "January", "February", etc.
+ *
+ * isFuture: Return boolean true if date is in the future (from when method is called)
+ *
+ * isToday: Return boolean true if the stored date is "today".
+ */
 
+var enhancedDate = (function () {
+    // private data variables
+    var myDate;
+    
+    return {
+        // public methods and properties
+        
+        setDate: function(dateItem) {
+            console.log("setDate");
+            if (dateItem === undefined) {
+                myDate = new Date();
+            } else {
+                myDate = new Date(dateItem);
+            }
+        },
+        
+        getDate: function(dateItem) {
+            console.log("getDate");
+            if (dateItem) {
+                return myDate;
+            } else {
+                return Date.UTC(myDate);
```

Comment:
myDate is a Date object, so when you pass it to Date.UTC it won't return what you want - Date.UTC wants a year and a month and, optionally, additional, more granular segments of time. It will return milliseconds, which is what you want your function to return here, but a simpler way to accomplish this would be to call `myDate.getTime( );`